cmd/clusterctl/client/cluster/cert_manager_test.go

```diff
@@ -41,15 +41,31 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	// Those values are dummy for test only
+	expectedHash    = "dummy-hash"
+	expectedVersion = "v0.11.2"
+
+	// This is needed for verify latest version and need to be aligned with current version
+	currentVersion = "v0.16.1"
+)
+
+func getHash() string {
+	yamlData, _ := manifests.Asset(embeddedCertManagerManifestPath)
+	return fmt.Sprintf("%x", sha256.Sum256(yamlData))
+}
+
 func Test_VersionMarkerUpToDate(t *testing.T) {
-	yaml, err := manifests.Asset(embeddedCertManagerManifestPath)
-	if err != nil {
-		t.Fatalf("Failed to get cert-manager.yaml asset data: %v", err)
+	pollImmediateWaiter := func(interval, timeout time.Duration, condition wait.ConditionFunc) error {
+		return nil
 	}
+	fakeConfigClient := newFakeConfig("")
+	cm, err := newCertManagerClient(fakeConfigClient, nil, pollImmediateWaiter)
 
-	actualHash := fmt.Sprintf("%x", sha256.Sum256(yaml))
 	g := NewWithT(t)
-	g.Expect(actualHash).To(Equal(embeddedCertManagerManifestHash), "The cert-manager.yaml asset data has changed, but embeddedCertManagerManifestVersion and embeddedCertManagerManifestHash has not been updated.")
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(cm.embeddedCertManagerManifestVersion).To(Equal(currentVersion), "The cert-manager.yaml asset data has changed, but embeddedCertManagerManifestVersion has not been updated")
+	g.Expect(cm.embeddedCertManagerManifestHash).To(Equal(getHash()), "The cert-manager.yaml asset data has changed, but embeddedCertManagerManifestHash has not been updated")
```

Comment:
Those error messages kind of confusing, because there is no more and embeddedCertManagerManifestVersion or hash I should take care of updating.

What about testing that we get a not nil Version and Hash, so we make sure that the newly updated cert-manager manifest is still compliant with the assumption we use while computing a value?
This will allow to drop currentVersion too...

Reply:
good point @fabriziopandini , I updated based on your suggestion ,thanks